### PR TITLE
fix(frontend): Fix searching for deleted users

### PR DESCRIPTION
**Pull request by shelegdmitriy**
Resolves: #598

### CODE REVIEW

File: frontend/src/libraries/explorer-wamp/contracts.ts [hunk 1]
```diff
@@ -56,8 +56,12 @@ export default class ContractsApi extends ExplorerApi {
     };
 
     try {
-      const codeHash = await this.queryCodeHash(id);
-      if (codeHash !== "11111111111111111111111111111111") {
+      // codeHash precheck is requered due to deleted accounts
+      let codeHash;
+      await this.queryCodeHash(id)
+        .then((it) => (codeHash = it))
+        .catch(() => {});
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/contracts.ts:63]: This should work, I believe. Please, double-check

```suggestion
      // codeHash does not exist for deleted accounts
      const codeHash = await this.queryCodeHash(id).catch(() => null);
```

File: frontend/src/libraries/explorer-wamp/accounts.ts [hunk 2]
```diff
@@ -197,6 +197,32 @@ export default class AccountsApi extends ExplorerApi {
     }
   }
 
+  async queryDbAccount(accountId: string): Promise<any> {
+    try {
+      let accountDetails;
+      if (this.dataSource === DATA_SOURCE_TYPE.INDEXER_BACKEND) {
+        accountDetails = this.call<any>("select:INDEXER_BACKEND", [
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/accounts.ts:204]: The returned value is not `accountDetails`. Please, be mindful of variable naming since misleading naming is worse than random names.

File: frontend/src/libraries/explorer-wamp/accounts.ts [hunk 3]
```diff
@@ -197,6 +197,32 @@ export default class AccountsApi extends ExplorerApi {
     }
   }
 
+  async queryDbAccount(accountId: string): Promise<any> {
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/accounts.ts:200]: Given that the only reason we added this new method is to check if the given account id has ever existed, let's implement the method `isAccountIndexed` and return `Promise<boolean>`

File: frontend/src/libraries/explorer-wamp/accounts.ts [hunk 4]
```diff
@@ -197,6 +197,32 @@ export default class AccountsApi extends ExplorerApi {
...
+    try {
+      let accountDetails;
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/accounts.ts:202]: There is no need for this variable, just return the result immediately.

File: frontend/src/libraries/explorer-wamp/accounts.ts [hunk 5]
```diff
@@ -197,6 +197,32 @@ export default class AccountsApi extends ExplorerApi {
...
+      if (this.dataSource === DATA_SOURCE_TYPE.INDEXER_BACKEND) {
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/accounts.ts:203]: I have already created the `legacy` branch of near-explorer, so we can strip all these `dataSource` checks in the master branch.

File: backend/package-lock.json [hunk 6]
```diff
@@ -1,1818 +1,8 @@
 {
   "name": "near-explorer-backend",
   "version": "0.0.1",
-  "lockfileVersion": 2,
```
Comment (frol) [inline backend/package-lock.json:4]: Please, revert the changes in the `backend/package-lock.json` since they are no-op and not relevant to this PR anyway.

File: frontend/src/libraries/explorer-wamp/contracts.ts (hunk 1 above)
Comment (shelegdmitriy) [reply frontend/src/libraries/explorer-wamp/contracts.ts]: Yes, you were right!

**Review by frol — APPROVED** [review]

File: frontend/src/libraries/explorer-wamp/accounts.ts [hunk 7]
```diff
@@ -197,6 +197,26 @@ export default class AccountsApi extends ExplorerApi {
     }
   }
 
+  async isAccountIndexed(accountId: string): Promise<boolean> {
+    try {
+      return this.call<any>("select:INDEXER_BACKEND", [
```
Comment (frol) [inline frontend/src/libraries/explorer-wamp/accounts.ts:202]:
```suggestion
      return await this.call<any>("select:INDEXER_BACKEND", [
```

**Review by frol — APPROVED** [review]
